hybrid/ddc_compartment_switching/shared.s

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) 2021 The CapableVMs "CHERI Examples" Contributors.
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+.global compartment_simple_fun
+
+.text
+.balign 4
+.global switch_compartment
+.type switch_compartment, "function"
+switch_compartment:
+    // int switch_compartment(int a, int b, void * stack, size_t size)
```

Comment:
Would I be silly if I said that we don't need `a, b` as arguments here (in the sense that they don't have anything to do with compartment switching as such)? Or is the intention here to model a slightly simplified mega-function `switch_compartment_and_call(new_stack, new_stack_size, function_f, function_f_arg_0, ... function_f_arg_n)`?

Reply:
I would have assumed `a` and `b` being the "content" but I am now trying to figure out how to show that "this works".

Reply:
I think at first it might be easiest to not have `a` and `b` in here at least at first (to keep it simple): when we switch compartments, the code can read whatever state it wants from the new heap.

Reply:
Actually there is no "new heap", because there's no easy way to temporarily ask `malloc` to allocate from a different area. The switcher just sets up a stack, for now. We could implement a simple allocator, but that would dominate the example.

Perhaps a second example could incorporate @jeremysinger's bump pointer allocator or similar.

Reply:
By the way, implementing a prototype-agnostic `switch_compartment_and_call` is much more difficult because it needs to do general-case argument marshalling.